geospaas/nansat_ingestor/managers.py

```diff
@@ -68,68 +68,119 @@ def get_or_create(self,
-            **options)
+        entry_id = n_metadata['entry_id']
+        if len(Dataset.objects.filter(entry_id__icontains=entry_id)) >= 1:
+            qds = Dataset.objects.filter(
+                entry_id__icontains=entry_id)  # query of found dataset
+            assert len(qds) == 1
+            ds = qds.first()  # this dataset (which is the first and the only one) must be updated
+            # ds. Here is the place for setting the metadata from the nansat that comes from the localfile
+            if (n_metadata['platform'] is not None and
+                    n_metadata['instrument'] is not None):
+                platform, _ = Platform.objects.get_or_create(
+                    json.loads(n_metadata['platform']))
+                instrument, _ = Instrument.objects.get_or_create(
+                    json.loads(n_metadata['instrument']))
+                specs = n_metadata.get('specs', '')
+                source, _ = Source.objects.get_or_create(platform=platform,
+                                                         instrument=instrument,
+                                                         specs=specs)
+                ds.source = source  # update source
+
+            if 'entry_title' in n_metadata and n_metadata['entry_title'] is not None:
+                # update entry_title
+                ds.entry_title = n_metadata['entry_title']
+            if 'summary' in n_metadata and n_metadata['summary'] is not None:
+                ds.summary = n_metadata['summary']  # update summary
+            if 'time_coverage_start' in n_metadata and n_metadata['time_coverage_start'] is not None:
+                # update time_coverage_start
+                ds.time_coverage_start = n_metadata['time_coverage_start']
+            if 'time_coverage_end' in n_metadata and n_metadata['time_coverage_end'] is not None:
+                # update time_coverage_end
+                ds.time_coverage_end = n_metadata['time_coverage_end']
+
+            if len(n.vrt.dataset.GetGCPs()) > 0:
+                n.reproject_gcps()
+                geolocation = GeographicLocation.objects.get_or_create(
+                    geometry=WKTReader().read(n.get_border_wkt(nPoints=n_points)))[0]
+                ds.geographic_location = geolocation
+
+            if 'data_center' in n_metadata and n_metadata['data_center'] is not None:
+                ds.data_center = DataCenter.objects.get_or_create(
+                    json.loads(n_metadata['data_center']))[0]
+            if 'gcmd_location' in n_metadata and n_metadata['gcmd_location'] is not None:
+                ds.gcmd_location = Location.objects.get_or_create(
+                    json.loads(n_metadata['gcmd_location']))[0]
+            if 'iso_topic_category' in n_metadata and n_metadata['iso_topic_category'] is not None:
+                ds.ISO_topic_category = ISOTopicCategory.objects.get_or_create(
+                    json.loads(n_metadata['iso_topic_category']))[0]
```

Comment:
Again repetition! It was so nicely DRIED in the previous code. But not anymore...